_shared/project/tests/unit/pytest_plugins/cookiecutter.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+{% if cookiecutter.get("_directory") == "pyramid-app" %}
+from pyramid import testing
+from pyramid.request import apply_request_extensions
+{% endif %}
+{% if cookiecutter.get("postgres") == "yes" %}
+
+from tests.factories.factoryboy_sqlalchemy_session import (
+    clear_factoryboy_sqlalchemy_session,
+    set_factorboy_sqlalchemy_session,
+)
+{% endif %}
+{% if cookiecutter.get("postgres") == "yes" %}
+
+
+@pytest.fixture
+def db(db_engine, db_sessionfactory):
+    """Return the SQLAlchemy database session.
+
+    This returns a session that is wrapped in an external transaction that is
+    rolled back after each test, so tests can't make database changes that
+    affect later tests.  Even if the test (or the code under test) calls
+    session.commit() this won't touch the external transaction.
+
+    This is the same technique as used in SQLAlchemy's own CI:
+    https://docs.sqlalchemy.org/en/20/orm/session_transaction.html#joining-a-session-into-an-external-transaction-such-as-for-test-suites
+    """
+    connection = db_engine.connect()
+    transaction = connection.begin()
+    session = db_sessionfactory(
+        bind=connection, join_transaction_mode="create_savepoint"
+    )
+    set_factorboy_sqlalchemy_session(session)
+
+    yield session
+
+    clear_factoryboy_sqlalchemy_session()
+    session.close()
+    transaction.rollback()
+    connection.close()
+{% endif %}
+{% if cookiecutter.get("_directory") == "pyramid-app" %}
+
+
+@pytest.fixture
+def pyramid_config(pyramid_settings):
+    with testing.testConfig(settings=pyramid_settings) as pyramid_config:
+        yield pyramid_config
+
+
+@pytest.fixture
+def pyramid_request(
+{% if cookiecutter.get("postgres") == "yes" %}
+    db,
+{% endif %}
+    pyramid_config,  # pylint:disable=unused-argument
+):
+    pyramid_request = testing.DummyRequest()
+    apply_request_extensions(pyramid_request)
+{% if cookiecutter.get("postgres") == "yes" %}
+    pyramid_request.db = db
+{% endif %}
+    return pyramid_request
+{% endif %}
```

Comment:
The cookiecutter will provide a `pyramid_request` fixture that will `pyramid_request.db` and `pyramid_request.tm` on it if the project uses Postgres. As with `pyramid_settings` the project can override this in its `conftest.py` and add more to it